packages/api/tests/lambdas/sf-event-sqs-to-db-records/test-index.js

```diff
@@ -361,16 +365,33 @@ test.serial('writeRecords() writes records to Dynamo and PostgreSQL', async (t)
   );
 });
 
-test('Lambda sends message to DLQ when writeRecords() throws an error', async (t) => {
+test.serial('Lambda sends message to DLQ when writeRecords() throws an error', async (t) => {
   const {
     handlerResponse,
-    sqsEvent,
   } = await runHandler({
     ...t.context,
     cumulusMessage: { fail: true },
   });
 
-  t.is(handlerResponse[0][1].body, sqsEvent.Records[0].body);
+  t.is(handlerResponse.batchItemFailures.length, 0);
+  const {
+    numberOfMessagesAvailable,
+    numberOfMessagesNotVisible,
+  } = await getSqsQueueMessageCounts(t.context.queues.deadLetterQueueUrl);
+  t.is(numberOfMessagesAvailable, 1);
+  t.is(numberOfMessagesNotVisible, 0);
+});
+
+test.serial('Lambda returns partial batch response to reprocess messages when getCumulusMessageFromExecutionEvent() throws an error', async (t) => {
+  const {
+    handlerResponse,
+  } = await runHandler({
+    ...t.context,
+    cumulusMessage: null,
```

Comment:
🤔  

This *does* result in the method under test throwing in the proxyquire mock, and validates the unit case, but does this change warrant a fully integrated unit test or a lambda-invocation integration test 
to show something like a message with  a granule that should succeed, one that failed on writeGranules and one that failed on getExecutionEvent do the right thing given a single execution?

Reply:
@Jkovarik I have addressed all your comments under [e71f482](https://github.com/nasa/cumulus/pull/3373/commits/e71f482603c5bd7e120c6ba08da45d33ad3094ad)